satellite/repair/checker/checker.go

```diff
@@ -163,7 +163,7 @@ func (checker *Checker) updateIrreparableSegmentStatus(ctx context.Context, poin
 
 	// we repair when the number of healthy pieces is less than or equal to the repair threshold
 	// except for the case when the repair and success thresholds are the same (a case usually seen during testing)
-	if numHealthy > redundancy.MinReq && numHealthy <= redundancy.RepairThreshold && numHealthy < redundancy.SuccessThreshold {
+	if numHealthy >= redundancy.MinReq && numHealthy <= redundancy.RepairThreshold && numHealthy < redundancy.SuccessThreshold {
```

Comment:
why we need this change? and rest in this file?

Reply:
Before repair with hashes, a segment was considered irreparable if it had `MinReq` or fewer pieces. This is because we needed at least `MinReq+1` pieces for the reed solomon error correction to work correctly. Now that we are verifying pieces with hashes before decoding into a segment, the minimum number exactly is sufficient to repair correctly.
The changes in this file ensure that we
1. Add to repair if we have exactly the minimum healthy number of pieces
2. Do not add to the irreparable db if we have exactly the minimum healthy number of pieces